lambdas/process-cri-callback/src/main/java/uk/gov/di/ipv/core/processcricallback/service/CriApiService.java

```diff
@@ -137,6 +138,11 @@ public HTTPRequest buildFetchAccessTokenRequest(
                                                             ConfigurationVariable.JWT_TTL_SECONDS)))
                                     .toEpochSecond(),
                             secureTokenHelper.generate());
+
+            if (signer instanceof KmsEs256Signer kmsEs256Signer) {
+                kmsEs256Signer.setKeyId(configService.getSigningKeyId());
+            }
```

Comment:
I think this is fine as-is, but if it's leaving you a bit unsatisfied, then a more robust approach might be to use a factory instead 🏭

i.e. create a `KmsSignerFactory` (which can cache the AWS KMS client), and then here we call: `kmsSignerFactory.getSigner(configService.getSigningKeyId())` to dynamically get an instance using the appropriate signing key.

(Alternatively, we could construct a new one every time - but that's much harder to test)

Reply:
Do you mean as an alternative to passing the signer in from the handler, and instead construct it in this class using a factory?
I'm not quite seeing how it helps with the need to use a different type of signer for the tests - hence why injecting it like we do is useful?

Reply:
The thing I don't like about this approach is that anybody using the `KmsEs256Signer` has to remember to call `setKeyId` before they can use it - which is a bit surprising.

If you inject a factory instead, then you can still inject a different version for the tests, but it also forces you to provide the key id in order to get a signer.

Reply:
Ah ok, I'm with you. Will fix it up.